src/std/misc/list.ss

```diff
@@ -11,49 +11,60 @@ package: std/misc
   snoc append1)
 
 (import
+  <host-runtime>
   :std/error :std/srfi/1)
 
 ;; This function transform a property list (k1 v1 k2 v2 ...) into
 ;; an association list ((k1 . v1) (k2 . v2) ...).
-;; NB: it uses equal? as the equality predicate for keys.
 (def (plist->alist plist)
   (let loop ((p plist))
     (match p
       ([k v . rest] (cons (cons k v) (loop rest)))
       ([] [])
-      ([_] (error "Odd number of elements in plist" plist)))))
+      (else (error "improper plist" plist)))))
 
 
 ;; Are the two lists of the same length. Note: diverges if either list is circular.
 (def (length=? x y) ;; Same as (= (length x) (length y))
-  (let ((nx? (null? x))
-        (ny? (null? y)))
+  (let ((nx? (not (pair? x)))
+        (ny? (not (pair? y))))
     (cond
      (nx? ny?)
      (ny? #f)
      (else (length=? (cdr x) (cdr y))))))
 
 ;; Is the list of a given length?
 (def (length=n? x n) ;; (= (length x) n)
-  (cond
-   ((null? x) (= 0 n))
-   ((= 0 n) #f)
-   (else (length=n? (cdr x) (- n 1)))))
+  (and (fixnum? n) (fx<= 0 n)
```

Comment:
if you are type checking the fixnum, then make it an error to pass anything else, as returning `#f` might be surprising behaviour.
But the check is redundant and implicit in the use of fx primitives, so probably better to omit it altogether.